osu.Game.Rulesets.Osu/Mods/OsuModSpinIn.cs

```diff
@@ -22,7 +22,7 @@ public class OsuModSpinIn : ModWithVisibilityAdjustment, IMutateApproachCircles
         public override double ScoreMultiplier => 1;
 
         // todo: this mod should be able to be compatible with hidden with a bit of further implementation.
-        public override Type[] IncompatibleMods => new[] { typeof(IMutateApproachCircles) };
+        public override Type[] IncompatibleMods => new[] { typeof(IRequiresApproachCircles), typeof(IHidesApproachCircles) };
```

Comment:
I could be fine with the rest of this but this does not seem correct - from a semantic point of view it does not feel like this mod should be incompatible with anything that can ever hide approach circles in the future.

I guess my question is: Is the implementation/nature of Spin In fundamentally incompatible with any future `IHidesApproachCircles` mod, or is it that just all other _current_ implementations of `IHidesApproachCircle` *happen* to not work with Spin In, but there's a possibility that one might in the future? Because if it's the latter, then this line is not right and it should name `OsuModHidden` and `OsuModObjectScaleTween` explicitly.

Reply:
Given that it's being incompatible with both `OsuModObjectScaleTween` and `OsuModHidden` which look to be completely unrelated from each other, I thought the former and decided to use that there, but could go the latter way... depends on why it's incompatible to begin with actually.

Reply:
A first step would be testing the combinations. If they don't look visually broken then there's no reason to make them incompatible.

Reply:
It seems they're actually incompatible, but for different reasons on each of them
- "Spin In" is incompatible with "Hidden" due to forcing the circle to stay completely opaque.
- "Spin In" is incompatible with "Grow"/"Deflate" since each of them apply to the object scale in different ways.

Will update the TODO comment.